### PR TITLE
timm integration

### LINKED ISSUE #172: Add Pytorch-Image-Models to backbones
## 🚀 Feature
https://github.com/rwightman/pytorch-image-models#models

### Motivation

The torch vision models are very limiting these models seem to perform best on kaggle challenges would be great to integrate them

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?
Fixes #172 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -17,3 +17,4 @@ sentencepiece>=0.1.95
 lightning-bolts==0.3.2  # todo: we shall align with proper release
 filelock  # comes with 3rd-party dependency
 pycocotools>=2.0.2 ; python_version >= "3.7"
+timm~=0.4.5
```
Comment (Borda) [inline requirements.txt:20]: this shall go to extras

File: flash/vision/backbones.py [hunk 2]
```diff
@@ -15,10 +15,14 @@
 
 import torchvision
 from pytorch_lightning.utilities import _BOLTS_AVAILABLE, rank_zero_warn
+from pytorch_lightning.utilities import _module_available
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from torch import nn as nn
 from torchvision.models.detection.backbone_utils import resnet_fpn_backbone
 
+if _module_available('timm'):
```
Comment (tchaton) [inline flash/vision/backbones.py:23]: Please, use _TIMM_AVAILABLE as for _BOLTS_AVAILABLE

File: flash/vision/backbones.py [hunk 3]
```diff
@@ -140,3 +147,21 @@ def torchvision_backbone_and_num_features(model_name: str, pretrained: bool = Tr
...
+def timm_backbone_and_num_features(model_name: str, pretrained: bool = True) -> Tuple[nn.Module, int]:
+    """
+    >>> timm_backbone_and_num_features('resnet18')  # doctest: +ELLIPSIS
+    (ResNet(...), 512)
+    >>> timm_backbone_and_num_features('mobilenetv3_large_100')  # doctest: +ELLIPSIS
+    (MobileNetV3(...), 1280)
+    """
+
+    if model_name in timm.list_models():
+        backbone = timm.create_model(model_name, pretrained=pretrained, num_classes=0,
```
Comment (tchaton) [inline flash/vision/backbones.py:161]: Why num_classes = 0 ?

File: flash/vision/backbones.py [hunk 4]
```diff
@@ -140,3 +147,21 @@ def torchvision_backbone_and_num_features(model_name: str, pretrained: bool = Tr
         return backbone, num_features
 
     raise ValueError(f"{model_name} is not supported yet.")
+
+
+def timm_backbone_and_num_features(model_name: str, pretrained: bool = True) -> Tuple[nn.Module, int]:
```
Comment (tchaton) [inline flash/vision/backbones.py:152]: Please, add a test.

File: flash/vision/backbones.py (hunk 2 above)
Comment (aniketmaurya) [reply flash/vision/backbones.py:23]: but _BOLTS_AVAILABLE comes from lightning. Should I define a new variable and then use it? @tchaton

File: flash/vision/backbones.py (hunk 3 above)
Comment (aniketmaurya) [reply flash/vision/backbones.py:161]: this method is suggested to create a model without a classification head in the timm docs. https://rwightman.github.io/pytorch-image-models/feature_extraction/#create-with-no-classifier-and-pooling

File: flash/vision/backbones.py (hunk 4 above)
Comment (aniketmaurya) [reply flash/vision/backbones.py:152]: [This test](https://github.com/aniketmaurya/lightning-flash/blob/issue_172-add_timm/flash/vision/backbones.py#L155) I have added.
Could you please refer me to where do I have to add the test for this function?

Comment (aniketmaurya) [reply flash/vision/backbones.py:152]: Do I have to add a backbone name for timm in the list? @aribornstein

Comment (aribornstein) [reply flash/vision/backbones.py:153]: @tchaton what do you prefer?

Comment (kaushikb11) [reply flash/vision/backbones.py:153]: I have added a test for it.

**Review by kaushikb11 — APPROVED** [review]

File: flash/vision/backbones.py [hunk 5]
```diff
@@ -140,3 +148,15 @@ def torchvision_backbone_and_num_features(model_name: str, pretrained: bool = Tr
...
+
+    if model_name in timm.list_models():
+        backbone = timm.create_model(model_name, pretrained=pretrained, num_classes=0, global_pool='')
+        num_features = backbone.num_features
+        return backbone, num_features
+
+    raise ValueError(
+        f"{model_name} is not supported in timm yet. https://rwightman.github.io/pytorch-image-models/models/"
```
Comment (carmocca) [inline flash/vision/backbones.py:161]: Maybe show `timm.list_models()`?

File: requirements/extras.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+timm~=0.4.5
```
Comment (carmocca) [inline requirements/extras.txt:1]: I think it's fine to do this instead

```suggestion
timm>=0.4.5
```

**Review by carmocca — APPROVED** [review]

File: flash/vision/backbones.py (hunk 5 above)
Comment (kaushikb11) [reply flash/vision/backbones.py:161]: It's a huge list. 347 models.

**Pull request merged**